### DISCUSSION TITLE
Hide navigation link only in Resources and not everywhere

### DISCUSSION THREAD

**Discussion by bendadaniel** [laravel/nova-issues#4368] [Q&A] [answered]
- Laravel Version: 9
- Nova Version: 4
- PHP Version: 8

Hello,
I want to move one of me resources to new Navigation section. I do not want to override whole navigation, because I want that normal other resources are automatically in Resources section. So I just prepend menu by:
```
$menu->prepend
(
  MenuSection::make('Supervisor tools', 
  [
       MenuItem::resource(FieldsPreviewResource::class),             
  ])
);
```

Work well, but now I have 2 same link in "Supervisor tools" section and in "Resources" section. When I disable navigation in resource, it hide both links. How can I just hide resource link in "Resources" section.
`public static $displayInNavigation = true;`

![Visualization](https://i.ibb.co/ZfqyPVD/BB5-CB4-FD-019-C-493-C-B248-F54-B45687-C8-C.jpg)

Thank you,
Daniel

**Comment by crynobone** [comment, marked as answer]
First set `public static $displayInNavigation = false;`

And then

```php
$menu->prepend(
  MenuSection::make('Supervisor tools', [
       MenuItem::resource(FieldsPreviewResource::class)
          ->canSee(fn ($request) => FieldsPreviewResource::authorizedToViewAny($request)),
  ])
);
```

**Reply by bendadaniel** [reply]
Hello @crynobone ,
you solution does not solved my problem, ofcourse I use "canSee" authorization, but my problem is that I can see 2 same navigation links. I want remove "Fields preview" from "Resources" section and keep it just in "Supervisor tools". Thank you for your help

**Reply by dmason30** [reply]
@bendadaniel He gave you the right solution. Add this to your `FieldPreviewResource`

`public static $displayInNavigation = false;`

**Reply by bendadaniel** [reply]
Aaah, I was adding canSee to MenuSection not to link, my fault. I am so sorry @crynobone, your solution works. Thank you so much for your time! :) (also thank you @dmason30)

**Comment by Fik21-i** [comment]
@bendadaniel hi, im still newbie. i just start doing laravel nova ,and my menu section just only have dashboard and resources . I have created couples of resource but it just go direct under resource menu. May i know , how you did to create other menu field like supervisor tools and statistic ? and where is the right path to put the code ?

**Reply by juse-less** [reply]
I recommend checking out [the documentation](https://nova.laravel.com/docs/4.0/installation.html), which has an entire page for [menus](https://nova.laravel.com/docs/4.0/customization/menus.html).

The section for customising the main menu can be found [here](https://nova.laravel.com/docs/4.0/customization/menus.html#customizing-the-main-menu).
You can also customise the user menu, in the top right-hand side corner, which can be found [here](https://nova.laravel.com/docs/4.0/customization/menus.html#customizing-the-user-menu).
Regardless of entire custom menus or single items, you can check out how to use the different types like [menu section](https://nova.laravel.com/docs/4.0/customization/menus.html#menu-sections), [menu group](https://nova.laravel.com/docs/4.0/customization/menus.html#menu-groups), [menu item](https://nova.laravel.com/docs/4.0/customization/menus.html#menu-items), etc, in the menu building section [here](https://nova.laravel.com/docs/4.0/customization/menus.html#building-menus).

There's a property, `$displayInNavigation`, in the Resource class that can also be used to hide it in the menu.
So simply adding `public static $displayInNavigation = false;` to the Resource class you want to hide, will automagically hide it.
But to make more custom stuff, you'll need to create a custom menu, as shown in my links above.

Apologies for just spamming links. It's a surprisingly wide subject, and smaller/more concrete questions is easier to answer. 🙂
